lib/puppet/provider/radius_server/nxapi.rb

```diff
@@ -0,0 +1,161 @@
+
+require 'cisco_node_utils' if Puppet.features.cisco_node_utils?
+begin
+  require 'puppet_x/cisco/autogen'
+rescue LoadError # seen on master, not on agent
+  # See longstanding Puppet issues #4248, #7316, #14073, #14149, etc. Ugh.
+  require File.expand_path(File.join(File.dirname(__FILE__), '..', '..', '..',
+                                     'puppet_x', 'cisco', 'autogen.rb'))
+end
+
+Puppet::Type.type(:radius_server).provide(:nxapi) do
+  desc 'The Cisco NXAPI provider for radius_server.'
+
+  confine feature: :cisco_node_utils
+  defaultfor operatingsystem: :nexus
+
+  mk_resource_methods
+
+  RADIUS_SERVER_PROPS = {
+    auth_port:           :auth_port,
+    acct_port:           :acct_port,
+    timeout:             :timeout,
+    retransmit_count:    :retransmit_count,
+    accounting_only:     :accounting,
+    authentication_only: :authentication,
+  }
+
+  def initialize(value={})
+    super(value)
+    @radius_server = Cisco::RadiusServer.radiusservers[@property_hash[:name]]
+    @property_flush = {}
+    debug 'Created provider instance of radius_server'
+  end
+
+  def self.get_properties(name, v)
+    debug "Checking instance, SyslogServer #{name}"
+
+    current_state = {
+      ensure:              :present,
+      name:                v.name,
+      auth_port:           v.auth_port ? v.auth_port : v.auth_port_default,
+      acct_port:           v.acct_port ? v.acct_port : v.acct_port_default,
+      timeout:             v.timeout ? v.timeout : -1,
+      retransmit_count:    v.retransmit_count ? v.retransmit_count : -1,
+      accounting_only:     v.accounting ? :true : :false,
+      authentication_only: v.authentication ? :true : :false,
+      key:                 v.key ? v.key : 'unset',
+      key_format:          v.key_format ? v.key_format : -1,
+    }
+
+    new(current_state)
+  end # self.get_properties
+
+  def self.instances
+    radiusservers = []
+    Cisco::RadiusServer.radiusservers.each do |name, v|
+      radiusservers << get_properties(name, v)
+    end
+
+    radiusservers
+  end
+
+  def self.prefetch(resources)
+    radius_servers = instances
+
+    resources.keys.each do |id|
+      provider = radius_servers.find { |instance| instance.name == id }
+      resources[id].provider = provider unless provider.nil?
+    end
+  end # self.prefetch
+
+  def munge_flush(val)
+    if val.is_a?(String) && val.eql?('unset')
+      nil
+    elsif val.is_a?(Integer) && val.eql?(-1)
+      nil
+    elsif val.is_a?(Symbol) && val.eql?(:true)
+      true
+    elsif val.is_a?(Symbol) && val.eql?(:false)
+      false
+    elsif val.is_a?(Symbol)
+      val.to_s
+    else
+      val
+    end
+  end
+
+  def validate
+    fail ArgumentError,
+         "This provider does not support the 'hostname' property. The namevar should be set to the IP of the Raidus Server" if @resource[:hostname]
+    fail ArgumentError,
+         "This provider does not support the 'group' property." if @resource[:group]
```

Comment:
With the code as written, if the user specifies multiple unsupported properties, only the first one will be reported as a failure - possibly requiring multiple iterations by the user to correct (fix one property, another one fails, fix that one, yet another fails). I wonder if there could be a cleaner way to report all failures in one go - something like:

```
UNSUPPORTED_PROPS = [:group, :deadtime, :vrf, ...]
invalid = UNSUPPORTED_PROPS & @resource.keys
fail ArgumentError, "This provider does not support the following properties: #{invalid}" unless invalid.empty?
```